src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -958,6 +958,11 @@ bool DiscoveryDataBase::delete_entity_of_change(
     return false;
 }
 
+bool DiscoveryDataBase::data_queue_empty()
+{
+    return data_queue_.Empty();
```

Comment:
I think this returns whether the Foreground queue is empty, where the Server pushes changes to the background queue. Foreground queue will always be empty after `process_data_queue()`. I think the easiest would be to check if both foreground and background are empty.
```suggestion
    return data_queue_.BothEmpty();
```